include/mbedtls/build_info.h

```diff
@@ -59,6 +59,13 @@
 #define inline __inline
 #endif
 
+/* Define `asm` for compilers which don't define it. */
+/* *INDENT-OFF* */
+#ifndef asm
```

Comment:
I don't think it's acceptable to mess with the definition of `asm` as a macro in a public header. This might conflict with other tricks that the application is pulling with `asm`.

Doing this in `library/common.h` would be acceptable.